### PR TITLE
More doctests for LinAlg

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: stdlib/LinearAlgebra/src/matmul.jl [hunk 1]
```diff
@@ -175,13 +175,45 @@ mul!(C::AbstractMatrix, A::AbstractVecOrMat, B::AbstractVecOrMat) = generic_matm
     rmul!(A, B)
 
 Calculate the matrix-matrix product ``AB``, overwriting `A`, and return the result.
+
+# Examples
+```jldoctest
+julia> using LinearAlgebra;
```
Comment (Sacha0) [inline stdlib/LinearAlgebra/src/matmul.jl]: Semicolon snuck in? :)

File: stdlib/LinearAlgebra/src/matmul.jl [hunk 2]
```diff
@@ -175,13 +175,45 @@ mul!(C::AbstractMatrix, A::AbstractVecOrMat, B::AbstractVecOrMat) = generic_matm
...
 rmul!(A, B)
 
 """
     lmul!(A, B)
 
 Calculate the matrix-matrix product ``AB``, overwriting `B`, and return the result.
+
+# Examples
+```jldoctest
+julia> using LinearAlgebra;
```
Comment (Sacha0) [inline stdlib/LinearAlgebra/src/matmul.jl]: Likewise here, sneaky semicolon? :)

**Review by Sacha0 — APPROVED** [review]
Looks great! Thanks @kshyatt! :)

File: stdlib/LinearAlgebra/src/matmul.jl (hunk 1 above)
Comment (fredrikekre) [reply stdlib/LinearAlgebra/src/matmul.jl]: I also don't think we have `using LinearAlgebra` in any other doctests?

**Comment by kshyatt** [comment]
I'll rebase if this fails!

File: stdlib/LinearAlgebra/src/LinearAlgebra.jl [hunk 3]
```diff
@@ -254,6 +254,29 @@ The reason for this is that factorization itself is both expensive and typically
...
+
+# Examples
+```jldoctest
+julia> A = [1 2.2 4; 3.1 0.2 3; 4 1 2];
+
+julia> X = [1; 2.5; 3];
+
+julia> Y = zero(X);
+
+julia> LinearAlgebra.ldiv!(Y, qr(A), X);
```
Comment (fredrikekre) [inline stdlib/LinearAlgebra/src/LinearAlgebra.jl]: Isn't `ldiv!` exported?

**Pull request merged**